### DISCUSSION TITLE
Can I use a SKR 1.4 Turbo for mini lathe CNC conversion?

### DISCUSSION THREAD

**Discussion by undingen** [grblHAL/core#112] [Q&A] [answered]
Hello
I'm currently researching how I can convert my mini lathe to CNC. GRBL-HAL is sounding very promising to me. If possible I would like to use a SKR 1.4 Turbo with TMC2209 (also have some external stepper controllers but forgot the name) to experiment with before buying new hardware. But one thing I definitely want is thread cutting support.

From the docs it seems like the HAL for LPC176x is not supporting spindle sync at the moment. 
Does anybody know if it's possible to implement it for this board and how hard it would be? I have seen the [Spindle sync developer reference](https://github.com/grblHAL/core/wiki/Spindle-sync-developer-reference) and quick googling shows that the chip has 4x 32-bit timers. I do know C but have no experience with GRBL-HAL or embedded development.

Also if possible I was thinking about using a CUI AMT102 encoder because it should be easy to add to my lathe and I can configure the number of steps and it has an index signal. Is it better to just build a custom encoder disk based on some hall effect / optical sensor?

**Comment by terjeio** [comment, marked as answer]
> From the docs it seems like the HAL for LPC176x is not supporting spindle sync at the moment.

True, and I have no plans to add it either as it is an old MCU with no FPU. There could be other limitations as well as the current code needs a counter that is clocked by the main encoder signal.

> Also if possible I was thinking about using a CUI AMT102 encoder 

It should be usable but may require level shifters for the signals.
